controllers/baremetal_config.go

```diff
@@ -0,0 +1,84 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	metal3iov1alpha1 "github.com/openshift/cluster-baremetal-operator/api/v1alpha1"
+)
+
+const (
+	provisioningNetworkManaged   = "Managed"
+	provisioningNetworkUnmanaged = "Unmanaged"
+	provisioningNetworkDisabled  = "Disabled"
+)
+
+func validateBaremetalProvisioningConfig(config *metal3iov1alpha1.Provisioning) error {
+	if config.Spec.ProvisioningNetwork == "" {
+		config.Spec.ProvisioningNetwork = provisioningNetworkManaged
+		if config.Spec.ProvisioningDHCPExternal {
+			config.Spec.ProvisioningNetwork = provisioningNetworkUnmanaged
+		}
+		glog.V(1).Infof("ProvisioningNetwork config not provided. Based on ProvisioningDHCPExternal setting it to: %s", config.Spec.ProvisioningNetwork)
+	}
+	switch config.Spec.ProvisioningNetwork {
+	case provisioningNetworkManaged:
+		glog.V(1).Info("Provisioning Network Managed")
+		return validateManagedConfig(config)
+	case provisioningNetworkUnmanaged:
+		glog.V(1).Info("Provisioning Network Unmanaged")
+		return validateUnmanagedConfig(config)
+	case provisioningNetworkDisabled:
+		glog.V(1).Info("Provisioning Network Disabled")
+		return validateDisabledConfig(config)
+	}
+	return nil
+}
+
+func validateManagedConfig(config *metal3iov1alpha1.Provisioning) error {
+	// All values must be provided in the provisioning CR
+	if config.Spec.ProvisioningInterface == "" ||
+		config.Spec.ProvisioningIP == "" ||
```

Comment:
We should also check whether the value specified is a valid IP. Something simple like:
e.g. https://github.com/openshift/installer/blob/b894e4c734366194b4773ca4bbcf90a249c2a7e4/pkg/types/baremetal/validation/platform.go#L126